src/framework/error.ts

```diff
@@ -0,0 +1,33 @@
+import { FastifyInstance } from 'fastify';
+
+import { logger } from './logging';
+
+export class JsonResponse extends Error {
+  constructor(
+    public statusCode: number,
+    public message: string,
+    public additionalFields: Record<string, unknown>,
+  ) {
+    super(message);
+    this.name = 'JsonResponse';
+  }
+  get response() {
+    return {
+      message: this.message,
+      ...this.additionalFields,
+    };
+  }
+}
+
+export const errorHandler: Parameters<FastifyInstance['setErrorHandler']>[0] = (
```

Comment:
Created a custom error handler to be able to throw custom fields in errors:

https://www.fastify.io/docs/latest/Reference/Reply/#async-await-and-promises

They support throwing errors with `statusCode` and `message` but you don't have the ability to return additional fields in the response.